frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/create-vm-wizard.tsx

```diff
@@ -0,0 +1,311 @@
+          return {
+            ...step,
+            name: TabTitleResolver[step.id],
+            enableNext: !isLocked && valid,
+            canJumpTo: !isLocked && prevStepValid && step.id !== VMWizardTab.RESULT,
+            hideBackButton: isLocked,
+            component: step.component,
+          };
+        })}
+      />
+    );
+  }
+}
+
+const wizardStateToProps = (state, { reduxID }) => ({
+  stepData: iGetCreateVMWizardTabs(state, reduxID),
+  // fetch data from store to detect and fire changes
+  ...[...DetectCommonDataChanges].reduce((acc, propName) => {
+    acc[propName] = iGetCommonData(state, reduxID, propName);
+    return acc;
+  }, {}),
+});
+
+const wizardDispatchToProps = (dispatch, props) => ({
+  onInitialize: () => {
+    dispatch(
+      vmWizardActions[ActionType.Create](props.reduxID, {
+        data: {
+          isCreateTemplate: props.isCreateTemplate,
+        },
+        dataIDReferences: props.dataIDReferences,
+      }),
+    );
+  },
+  onCommonDataChanged: (commonData: CommonData, changedCommonData: ChangedCommonData) => {
+    dispatch(
+      vmWizardActions[ActionType.UpdateCommonData](props.reduxID, commonData, changedCommonData),
+    );
+  },
+  onResultsChanged: (results, isValid) => {
+    dispatch(vmWizardActions[ActionType.SetResults](props.reduxID, results, isValid));
+  },
+  onClose: () => {
+    if (props.onClose) {
+      props.onClose();
+    }
+    dispatch(vmWizardActions[ActionType.Dispose](props.reduxID, props));
+  },
+});
+
+export const CreateVMWizard = connect(
+  wizardStateToProps,
+  wizardDispatchToProps,
+)(CreateVMWizardComponent);
+
+export const CreateVMWizardPageComponent: React.FC<CreateVMWizardPageComponentProps> = (props) => {
+  const {
+    reduxID,
+    match: {
+      params: { ns: activeNamespace },
+    },
+  } = props;
+  const resources = [
+    getResource(VirtualMachineModel, {
+      namespace: activeNamespace,
+      prop: VMWizardProps.virtualMachines,
+    }),
+    getResource(TemplateModel, {
+      namespace: activeNamespace,
+      prop: VMWizardProps.userTemplates,
+      matchLabels: { [TEMPLATE_TYPE_LABEL]: TEMPLATE_TYPE_VM },
+    }),
+    getResource(TemplateModel, {
+      namespace: 'openshift',
+      prop: VMWizardProps.commonTemplates,
+      matchLabels: { [TEMPLATE_TYPE_LABEL]: TEMPLATE_TYPE_BASE },
+    }),
+    getResource(NetworkAttachmentDefinitionModel, {
+      namespace: activeNamespace,
+      prop: VMWizardProps.networkConfigs,
+    }),
+    getResource(StorageClassModel, { prop: VMWizardProps.storageClasses }),
+    getResource(PersistentVolumeClaimModel, {
+      namespace: activeNamespace,
+      prop: VMWizardProps.persistentVolumeClaims,
+    }),
+    getResource(DataVolumeModel, {
+      namespace: activeNamespace,
+      prop: VMWizardProps.dataVolumes,
+    }),
+  ];
+
+  const dataIDReferences = resources.reduce((acc, resource) => {
```

Comment:
firehose injects `reduxIDs` props so you dont have to compute them yourself

Reply:
That is true, but it is not enough. We need to pair the reference path with the prop. Please see:

```tsx
 acc[resource.prop] = ['k8s', makeReduxID(resource.model, query)];
```

This is done to get our desired prop even if the query and redux id changes (e.g. namespace change).